src/pip/_internal/commands/debug.py

```diff
@@ -63,6 +63,14 @@ def create_vendor_txt_map():
     return dict(line.split('==', 1) for line in lines)  # type: ignore
 
 
+def create_debundle_txt_map():
+    # type: () -> Dict[str, str]
+    wheels = [fn for fn in os.listdir(pip._vendor.WHEEL_DIR)]
```

Comment:
This will also catch non-wheel files (which are likely to exist if the wheels are in `pip/_vendor`). We have a `Wheel` helper in `pip._internal.models.wheel` that can help distinguish those cases and get the `name` and `version`:

```python
def _vendor_wheels():
    for fn in os.listdir(pip._vendor.WHEEL_DIR):
        try:
            yield Wheel(fn)
        except InvalidWheelFilename:
            pass


def create_debundle_txt_map():
    return {wheel.name: wheel.version for wheel in _vendor_wheels()}
```

Reply:
Same point as in https://github.com/pypa/pip/pull/8328#pullrequestreview-484457883
`WHEEL_DIR` is currently not the only mechanism used by distributions to debundle pip.